### PR TITLE
feat: Implemented route and service for us0004

### LINKED ISSUE #10: US - 0004
Criação da rota com listagem de todos os vendedores que determinado usuário segue e implementação do serviço.

### PR DESCRIPTION
**Pull request by Victor-Planas**
feat: GenericResponseDTO created
feat: Route and service for getting a list of seller that a given user follows
fix: check if customer already follows a seller

### CODE REVIEW

File: src/main/java/br/com/mercadolivre/bootcampw2/grupo11/socialmeli/controllers/UsersController.java [hunk 1]
```diff
@@ -45,11 +49,14 @@ public UserDTO getUserInfo(@PathVariable @Min(0) Integer id) {
...
-    @ResponseStatus(HttpStatus.OK)
-    @Operation(description = "Fetch a user by id, works for customers and sellers")
-    public void follow(@PathVariable @Min(0) Integer userId, @PathVariable @Min(0) Integer userIdToFollow) {
-        usersService.follow(userId, userIdToFollow);
+    @GetMapping("/{userId}/followed/list")
+    public UserFollowingListDTO getSellersFollowedByUser(@PathVariable @Min(0) Integer userId){
+        return usersService.getFollowingList(userId);
     }
 
+    @PostMapping("/{userId}/follow/{userIdToFollow}")
```
Comment (CarlosFeldmann) [inline src/main/java/br/com/mercadolivre/bootcampw2/grupo11/socialmeli/controllers/UsersController.java:57]: Missing documentation for the swagger
@Operation
@ResponseStatus

File: src/main/java/br/com/mercadolivre/bootcampw2/grupo11/socialmeli/controllers/UsersController.java [hunk 2]
```diff
@@ -45,11 +49,14 @@ public UserDTO getUserInfo(@PathVariable @Min(0) Integer id) {
         return usersService.getUserInfo(id);
     }
 
-    @PostMapping("/{userId}/follow/{userIdToFollow}")
-    @ResponseStatus(HttpStatus.OK)
-    @Operation(description = "Fetch a user by id, works for customers and sellers")
-    public void follow(@PathVariable @Min(0) Integer userId, @PathVariable @Min(0) Integer userIdToFollow) {
-        usersService.follow(userId, userIdToFollow);
+    @GetMapping("/{userId}/followed/list")
```
Comment (CarlosFeldmann) [inline src/main/java/br/com/mercadolivre/bootcampw2/grupo11/socialmeli/controllers/UsersController.java:49]: Missing documentation

**Review by CarlosFeldmann — CHANGES_REQUESTED** [review]
La vem o homi macaco correndo atrás de mim 🦧

**Review by CarlosFeldmann — APPROVED** [review]
Perfeição 🦧

**Review by Giovannapls — APPROVED** [review]
🤖 dale

**Review by lmarizmeli — APPROVED** [review]

**Review by lmarizmeli — APPROVED** [review]
👽

**Pull request merged**